File: hw/chalan/u3/hw29Singleton/photo_studio_system/controller/photographer_controller.py
from model.photographer import Photographer
from utils.json_file_util import JsonFileUtil
from utils.mongo_db_util import MongoDBUtil

class PhotographerController: #sing

    def register(self, name, specialty, experience_text, rate_text):
        try:
            experience = int(experience_text)
            hourly_rate = float(rate_text)
        except ValueError:
            raise ValueError("Values must be numeric")

        photographer = Photographer(name, specialty, experience, hourly_rate)

        # 
        JsonFileUtil().save(photographer)
        MongoDBUtil().save(photographer)

    def get_photographers(self):
        # Delegacion
        return MongoDBUtil().get_all()

File: hw/chalan/u3/hw29Singleton/photo-system-js/controller/PhotographerController.js
const Photographer = require("../model/Photographer");
const JsonFileUtil = require("../utils/JsonFileUtil");
const MongoDbUtil = require("../utils/MongoDbUtil");

class PhotographerController { // 

    async register(name, specialty, experienceText, rateText) {
        const experience = parseInt(experienceText);
        const hourlyRate = parseFloat(rateText);

        if (isNaN(experience) || isNaN(hourlyRate)) {
            throw new Error("Values must be numeric");
        }

        const photographer = new Photographer(name, specialty, experience, hourlyRate);

        // 
        JsonFileUtil.getInstance().save(photographer);
        await MongoDbUtil.getInstance().save(photographer);
    }

    async getPhotographers() {
        return await MongoDbUtil.getInstance().getAll(); 
    }

    
    async findPhotographer(name) {
        return await MongoDbUtil.getInstance().getOne(name); 
    }
}

module.exports = PhotographerController;